plugins/woocommerce/client/blocks/assets/js/blocks/product-details/utils.ts

```diff
@@ -0,0 +1,128 @@
+/**
+ * External dependencies
+ */
+import {
+	PartialProduct,
+	ProductDimensions,
+	ProductProductAttribute,
+} from '@woocommerce/data';
+import { __ } from '@wordpress/i18n';
+
+export const isAdditionalProductDataEmpty = (
+	product: PartialProduct
+): boolean => {
+	const isWeightEmpty = ( value: string | undefined ) => {
+		return ! value || value.length === 0;
+	};
+
+	const isDimensionsEmpty = ( value: ProductDimensions | undefined ) => {
+		return (
+			! value ||
+			Object.values( value ).every(
+				( val ) => ! val || val.trim() === ''
+			)
+		);
+	};
+
+	const isAttributesEmpty = (
+		value: ProductProductAttribute[] | undefined
+	) => {
+		return ! value || value.length === 0;
```

Comment:
This seems to be the same as `isWeightEmpty()`. Deduplicate?

Reply:
Yeah, the implementation is the same, but the argument differs. I’m fine with a shared function for both attributes, but this is one of the rare cases where duplication improves clarity. Also, these functions are only scoped to this file.